### ISSUE TITLE
State of maintainership?

### ISSUE THREAD

**Issue by Ybalrid** [rust-openvr/rust-openvr#57]
Hello 👋 

I've been reading the discussions surrounding this rust library, and, as far as I understand, there are no plans from the org to do any significant work to these OpenVR bindings. (c.f.: https://github.com/rust-openvr/rust-openvr/pull/47#issuecomment-1025366777)

Currently, this crate is helpful but technically needs to be updated with Valve's current version of the OpenVR SDK. It also needs to be completed (lacking support for a number of still useful graphics API. e.g., Direct3D 11, see #56). There is also a *large number of forks* with potentially useful patches that are not contributing upstream (the joys of free and open source software in the era of GitHub, I guess 😉) 

I understand that OpenXR is available, but some people want to concentrate their efforts there. But OpenVR is still the main API to access several functionalities of SteamVR that *are not provided by the OpenXR 1.0 standard* notably:

- **SteamVR Drivers** : The ability to write a plugin that extends the functionalities of SteamVR, for example: targeting different hardware for rendering and inputs.
- **SteamVR Overlay applications** Currently, there is no real solution to do native OpenXR overlays. This is a whole class of VR applications that cannot use OpenXR and probably will not for a long time. The [XR_EXTX_overlay](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XR_EXTX_overlay) extension is only provisional (meaning: experimental) and is not actually implemented by *any* OpenXR runtime currently. The only implementation is an [extremely hacky API Layer by LunarG](https://github.com/LunarG/OpenXR-OverlayLayer) that isn't maintained actively. 
- **SteamVR Background application** There is no good way to create an OpenXR application that only receives inputs and does no do rendering. This is partially doable if you have support for the (non multi-vendor) [XR_MDN_headless](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XR_MND_headless) extension for example. This does not map 1:1 with OpenVR-provided functionalities...

Thus, IMHO the idea that OpenVR is not useful anymore is incorrect. I understand that the maintainer's interest might have switched from using OpenVR to instead using OpenXR, and that's perfectly fine! 🙂

I'd like to have some clarity on the state of the future of this crate and eventually offer some help in keeping it up-to-date / implementing missing functionalities... 🤔 

(Context: we're looking at Rust at my day job for a number of things. I work for a company in the XR space that still heavily uses SteamVR and OpenVR in its products for a number of the reasons listed above...)

**Comment by Ralith** [comment]
This crate is not maintained. It sounds like you have some compelling applications for it; perhaps you'd like to take on maintainership yourself? I don't have credentials for that, but hopefully whoever does can be tracked down.

I also encourage you/your employer to work with Khronos directly to try to get these gaps addressed in OpenXR itself as a more sustainable long-term solution.

**Comment by Ybalrid** [comment]
> This crate is not maintained. It sounds like you have some compelling applications for it; perhaps you'd like to take on maintainership yourself? I don't have credentials for that, but hopefully whoever does can be tracked down.

I figured that out, and that issue is kind of a proposal for it
 
> I also encourage you/your employer to work with Khronos directly to try to get these gaps addressed in OpenXR itself as a more sustainable long-term solution.

Yes, We (LIV Inc.) are part of the OpenXR working group. I cannot comment on the internals of the discussions that occur there. However, you can probably imagine that committee/standardization work is a very long process; And reaching a consensus with all the parties involved is not always an easy thing. 🙂

**Comment by Ybalrid** [comment]
> Hello @Ybalrid Sorry for the late response, looks like I only receive notifications when directly mentioned. So generally like you mentioned I'm no longer in need of the crate myself and hence I didn't spend any time on it. I also agree that OpenVR is still a viable choice for some projects despite the existence of OpenXR, so if you want to take ownership & maintance of this crate I'm happy to provide you all permissions (Github Org + crates.io)
> 
> So just to confirm again you are interessted in becoming a maintainer correct?

Yes I am, I'll be glad to take care of the OpenVR Rust bindings!🙂

**Comment by reneeichhorn** [comment]
Thanks a lot for taking care 👍🏻 
You should have got two invitation, with that you should be also able to publish to crates.io. Let me know if you need anything else

**Comment by Ybalrid** [comment]
Thanks!

**Comment by Ybalrid** [comment]
@reneeichhorn Hello again, I seem to not have the rights on the other repositories from the org? I may have to get started getting updates done on the `sys` crate, as I want to bring this up to speed with the most recent OpenVR version. 

Also, that struct alignment issue regarding padding with bindgen is *quite strange*, I am very curious about what is going on there...

**Comment by Ybalrid** [comment]
Also, I am pretty sure we should remove, or change the avatar/logo of this org to something else, as I recall [something about recent trademark policy changes](https://foundation.rust-lang.org/policies/logo-policy-and-media-guide/)

**Comment by reneeichhorn** [comment]
You should have full permission now. And yes agree on the logo, removed it for now

**Comment by kaitlynkittyy** [comment]
Is the project still planned to be maintained? The last commit shows as having been about 2 years ago.

**Comment by hnorkowski** [comment]
@Ybalrid Any updates?

**Comment by Ybalrid** [comment]
Yes, this create and the associated -sys one, are as declared, `passively maintained` by a few members of our team at LIV Inc. It recently got a few updates (few missing function calls, and updated to the latest version of OpenVR SDK from Valve).

If there any part of the OpenVR API that is missing or anything, do not hesitate to open a issue or submit a pull request.

With that said, I am closing this one now
